crates/supervisor/core/src/supervisor.rs

```diff
@@ -215,6 +219,34 @@ impl Supervisor {
         }
         Ok(())
     }
+
+    fn is_interop_enabled(&self, chain_id: ChainId, timestamp: u64) -> bool {
+        self.config
+            .rollup_config_set
+            .get(chain_id)
+            .map(|cfg| cfg.is_post_interop(timestamp))
+            .unwrap_or(false) // if config not found, return false
+    }
+
+    fn verify_safety_level(
+        &self,
+        chain_id: ChainId,
+        block: &BlockInfo,
+        safety: SafetyLevel,
+    ) -> Result<(), SupervisorError> {
+        // check block exists on the derivation storage
+        self.database_factory
+            .get_db(chain_id)?
+            .derived_to_source(BlockNumHash { number: block.number, hash: block.hash })?;
+
+        let head_ref = self.database_factory.get_db(chain_id)?.get_safety_head_ref(safety)?;
+
+        if head_ref.number < block.number {
+            return Err(SupervisorError::from(SuperchainDAError::ConflictingData));
```

Comment:
nice that you're making use of `SuperchainDAError` already! assuming Go impl returns the same error if this condition is met

Reply:
Yes.